bootstrap/tracing.go

```diff
@@ -73,7 +73,7 @@ func (b *Bootstrap) ddResourceName() string {
 // abstraction so the agent can support multiple libraries if needbe.
 func (b *Bootstrap) startTracingDatadog(ctx context.Context) (tracetools.Span, context.Context, stopper) {
 	opts := []tracer.StartOption{
-		tracer.WithServiceName("buildkite_agent"),
+		tracer.WithServiceName("buildkite-agent"),
```

Comment:
👍🏼

Will this affect existing OpenTracing/Datadog usage?

Reply:
nah, this only affects opentel. datadog initialision happens in the `startTracingDatadog` method above, which keeps the underscored name. the two methods are mutually exclusive, only one of them can (should?) be called per bootstrap run.